tests/server/auth/test_auth.py

```diff
@@ -54,6 +56,44 @@ def test_authenticate(client, monkeypatch):
         client.search_experiments()
 
 
+def _mlflow_search_experiments_rest(base_uri, headers):
+    response = requests.post(
+        f"{base_uri}/api/2.0/mlflow/experiments/search",
+        headers=headers,
+        json={
+            "max_results": 100,
+        },
+    )
+    response.raise_for_status()
+    return response
+
+
+@pytest.mark.parametrize(
+    "client",
+    [{"MLFLOW_AUTH_CONFIG_PATH": "tests/server/auth/fixtures/jwt_auth.ini"}],
+    indirect=True,
+)
+def test_authenticate_jwt(client):
+    # unauthenticated
+    with pytest.raises(requests.HTTPError, match=r"401 Client Error: UNAUTHORIZED") as e:
+        _mlflow_search_experiments_rest(client.tracking_uri, {})
+    assert e.value.response.status_code == 401  # Unauthorized
+
+    # authenticated
+    username, password = create_user(client.tracking_uri)
+    headers = {
+        "Authorization": f'Bearer {jwt.encode({"username": username}, "secret", algorithm="HS256")}'
+    }
+    _mlflow_search_experiments_rest(client.tracking_uri, headers)
+
+    # invalid token
+    bearer_token = jwt.encode({"username": username}, "invalid", algorithm="HS256")
+    headers = {"Authorization": f"Bearer {bearer_token}"}
+    with pytest.raises(requests.HTTPError, match=r"401 Client Error: UNAUTHORIZED") as e:
+        _mlflow_search_experiments_rest(client.tracking_uri, headers)
+    assert e.value.response.status_code == 401  # Unauthorized
```

Comment:
can we just mock `tests.server.auth.jwt_auth:authenticate_request` (or, better, some other function name in the config file) and assert that it has been called? it looks a bit weird to test custom logic that is not in the `mlflow` package